src/app/modules/sgid/sgid.controller.ts

```diff
@@ -58,10 +66,40 @@ export const handleLogin: ControllerHandler<
     res.cookie('isLoginError', true)
     return res.redirect(target)
   }
+  if (form.authType === FormAuthType.SGID_MyInfo) {
+    const jwtResult = await SgidService.retrieveAccessToken(code).andThen(
+      (data) => SgidService.createSgidMyInfoJwt(data.accessToken),
+    )
+
+    if (jwtResult.isErr()) {
+      logger.error({
+        message: 'Error while handling login via sgID (MyInfo)',
+        meta,
+        error: jwtResult.error,
+      })
+      res.cookie('isLoginError', true)
+      return res.redirect(target)
+    }
+
+    const { maxAge, jwt } = jwtResult.value
+    res.cookie(SGID_MYINFO_COOKIE_NAME, jwt, {
+      maxAge,
+      httpOnly: true,
+      sameSite: 'lax', // Setting to 'strict' prevents Singpass login on Safari, Firefox
+      secure: !config.isDev,
+      ...SgidService.getCookieSettings(),
+    })
+    return res.redirect(target)
+  }
 
   const jwtResult = await SgidService.retrieveAccessToken(code)
     .andThen((data) => SgidService.retrieveUserInfo(data))
-    .andThen(({ data }) => SgidService.createJwt(data, rememberMe))
+    .andThen(({ data }) =>
+      SgidService.createSgidSingpassJwt(
+        { 'myinfo.nric_number': data[SGID_MYINFO_NRIC_NUMBER_SCOPE] },
+        rememberMe,
+      ),
```

Comment:
Would this be neater instead of hardcoding
```
SgidService.createSgidSingpassJwt(
        { [SGID_MYINFO_NRIC_NUMBER_SCOPE]: data[SGID_MYINFO_NRIC_NUMBER_SCOPE] },
        rememberMe,
      ),
```

Reply:
Thanks! I completely forgot that's how you use variables as keys in JS!